10.0/s2i/bin/assemble

```diff
@@ -150,6 +150,64 @@ function ismgmtup() {
   return 1
 }
 
+# insert settings for HTTP proxy into settings.xml if supplied
+function configure_proxy() {
+  if [ -n "$HTTP_PROXY_HOST" -a -n "$HTTP_PROXY_PORT" ]; then
+    xml="<proxy>\
+         <id>genproxy</id>\
+         <active>true</active>\
+         <protocol>http</protocol>\
+         <host>$HTTP_PROXY_HOST</host>\
+         <port>$HTTP_PROXY_PORT</port>"
+    if [ -n "$HTTP_PROXY_USERNAME" -a -n "$HTTP_PROXY_PASSWORD" ]; then
+      xml="$xml\
+         <username>$HTTP_PROXY_USERNAME</username>\
+         <password>$HTTP_PROXY_PASSWORD</password>"
+    fi
+    if [ -n "$HTTP_PROXY_NONPROXYHOSTS" ]; then
```

Comment:
i guess since it matches what EAP uses, we can live with it, but NO_PROXY probably would have been a better env var to use for this.